File: src/main/java/com/circleescape/server/scoreboard/GreetingController.java
package com.circleescape.server.scoreboard;

import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;

import java.util.Set;

@Controller
public class GreetingController {

    @GetMapping("/greeting")
    public String greeting(
            @RequestParam(name = "name", required = false, defaultValue = "World") String name,
            Model model) {
        model.addAttribute("name", name);

        Set<Score> scores = Set.of(
                new Score("alice", 0.6f),
                new Score("bob", 0.9f));

        model.addAttribute("scoreTests", scores);

        return "greeting";
    }

}

File: src/main/kotlin/com/circleescape/server/scoreboard/GreetingController.kt
package com.circleescape.server.scoreboard

import com.circleescape.server.scoreboard.data.ScoreRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestParam

@Controller
class GreetingController {
    @Autowired
    var scoreRepository: ScoreRepository? = null
    @GetMapping("/greeting")
    fun greeting(
        @RequestParam(name = "name", required = false, defaultValue = "World") name: String?,
        model: Model
    ): String {
        model.addAttribute("name", name)
        return "greeting"
    }
}